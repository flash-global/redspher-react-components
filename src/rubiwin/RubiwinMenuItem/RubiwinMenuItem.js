import React from 'react'
import PropTypes from 'prop-types'
import MenuItem from '@material-ui/core/MenuItem'
import style from './rubiwinMenuItem.module.scss'

const RubiwinMenuItem = ({ children, className, component, ...props }) => (
  <MenuItem
    className={className}
    component={component}
    classes={{ root: style.root, selected: style.selected }}
    {...props}
  >
    {children}
  </MenuItem>
)

RubiwinMenuItem.propTypes = {
  /** children */
  children: PropTypes.node,
  /** add a class to children */
  className: PropTypes.string,
  /** The component used for the root node. Either a string to use a HTML element or a component. */
  component: PropTypes.string
}

export default RubiwinMenuItem
