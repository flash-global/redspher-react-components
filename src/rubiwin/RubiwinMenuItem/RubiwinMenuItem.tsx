import MenuItem from '@mui/material/MenuItem';
import { ExtendButtonBase } from '@mui/material/ButtonBase';
import { MenuItemTypeMap } from '@mui/material/MenuItem/MenuItem';

interface IrubiwinMenuItem extends ExtendButtonBase<MenuItemTypeMap> {
  children?: Node;
  className?: string;
}

/**
 * This is a Mui RubiwinMenuItem branded for Rubiwin
 *
 * Demos:
 * - [Menus](https://mui.com/components/menus/)
 *
 * API:
 * - [MenuItem API](https://mui.com/api/menu-item/)
 * - inherits [ButtonBase API](https://mui.com/api/button-base/)
 */
const RubiwinMenuItem = ({
  children,
  className,
  ...props
}: IrubiwinMenuItem) => (
  <MenuItem className={className} {...props}>
    {children}
  </MenuItem>
);

export default RubiwinMenuItem;
