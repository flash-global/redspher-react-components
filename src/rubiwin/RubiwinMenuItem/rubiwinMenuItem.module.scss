@import '../rubiwin-constant';

.root {
  &:hover {
    background-color: rgba($primaryColor, 0.3);
  }
}

.selected {
  background-color: rgba($primaryColor, 0.8) !important;
  color: white;
}
