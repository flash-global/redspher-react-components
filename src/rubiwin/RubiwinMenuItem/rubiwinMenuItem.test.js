import React from 'react';
import renderer from 'react-test-renderer';
import RubiwinMenuItem from './RubiwinMenuItem';

it('renders correctly', () => {
  const tree = renderer
    .create(<RubiwinMenuItem className='testClass' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders as a div', () => {
  const tree = renderer
    .create(<RubiwinMenuItem className='testClass' component='div' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
