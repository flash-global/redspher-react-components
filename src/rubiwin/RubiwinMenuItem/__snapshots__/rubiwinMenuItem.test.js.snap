// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders as a div 1`] = `
<div
  className="root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root testClass css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
/>
`;

exports[`renders correctly 1`] = `
<li
  className="root MuiMenuItem-root MuiMenuItem-gutters MuiButtonBase-root testClass css-kk1bwy-MuiButtonBase-root-MuiMenuItem-root"
  onBlur={[Function]}
  onContextMenu={[Function]}
  onDragLeave={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  role="menuitem"
  tabIndex={-1}
/>
`;
